File: back-end/php/task_management/api/controllers/presence_hours_controller.php
<?php

class presence_hours_controller extends controller_singletone {

    //post
    //route=addPresenceHours
    public function add_presence_hours($params) {
        return $this->service->add_presence_hours($params);
    }

    //put
    //route=editPresenceHours
    public function edit_presence_hours($params) {
        return $this->service->edit_presence_hours($params);
    }

    //get
    //route=getPresenceStatusPerWorkers
    public function get_presence_status_per_workers($params) {
        return $this->service->get_presence_status_per_workers($params['teamLeaderId']);
    }

    //get
    //route=getPresenceStatusPerProjects
    public function get_presence_status_per_projects($params) {
        return $this->service->get_presence_status_per_projects($params['workerId']);
    }

    //get
    //route=getPresenceHoursSum
    public function get_presence_hours_sum($params) {
        return $this->service->get_presence_hours_sum($params['projectId'], $params['workerId']);
    }

    //get
    //route=getPresenceHours
    public function get_presence_hours($params) {
        return $this->service->get_presence_hours($params['projectId'], $params['workerId']);
    }

}

File: back-end/c#/TaskManagement/API/Controllers/PresenceHoursController.cs
using BOL;
using BLL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using System.Web.Http.Cors;

namespace API.Controllers
{
    [EnableCors("*", "*", "*")]
    [RoutePrefix("api/presenceHours")]
    public class PresenceHoursController : BaseController
    {

        [HttpPost]
        [Route("addPresenceHours")]
        public HttpResponseMessage AddPresenceHours(PresenceHours newPresenceHours)
        {
            try
            {
                if (ModelState.IsValid)
                {
                    int id = PresenceHoursService.AddPresenceHours(newPresenceHours);
                    return Request.CreateResponse(HttpStatusCode.Created, id);
                }
                List<string> errors = GetErrorList(ModelState.Values.ToList());

                return Request.CreateResponse(HttpStatusCode.BadRequest, errors);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message);
            }

        }

        [HttpPut]
        [Route("editPresenceHours")]
        public HttpResponseMessage EditPresenceHours(PresenceHours presenceHours)
        {
            try
            {
                bool edited = PresenceHoursService.EditPresenceHours(presenceHours);
                return Request.CreateResponse(HttpStatusCode.OK, edited);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message);
            }
        }

        [HttpGet]
        [Route("getPresenceStatusPerWorkers")]
        public HttpResponseMessage GetPresenceStatusPerWorkers(int teamLeaderId)
        {
            try
            {

                List<dynamic> presenceStatusList = PresenceHoursService.GetPresenceStatusPerWorkers(teamLeaderId);
                return Request.CreateResponse(HttpStatusCode.OK, presenceStatusList);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("getPresenceStatusPerProjects")]
        public HttpResponseMessage GetPresenceStatusPerProjects(int workerId)
        {
            try
            {

                List<dynamic> presenceStatusList = PresenceHoursService.GetPresenceStatusPerProjects(workerId);
                return Request.CreateResponse(HttpStatusCode.OK, presenceStatusList);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("getPresenceHoursSum")]
        public HttpResponseMessage GetPresenceHoursSum(int projectId, int workerId)
        {
            try
            {

                object presenceSum = PresenceHoursService.GetPresenceHoursSum(projectId,workerId);
                return Request.CreateResponse(HttpStatusCode.OK, presenceSum);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }
    }
}
